wpilibj/src/main/java/edu/wpi/first/wpilibj/AddressableLEDBuffer.java

```diff
@@ -27,9 +40,41 @@ public AddressableLEDBuffer(int length) {
    */
   @Override
   public void setRGB(int index, int r, int g, int b) {
-    m_buffer[index * 4] = (byte) b;
-    m_buffer[(index * 4) + 1] = (byte) g;
-    m_buffer[(index * 4) + 2] = (byte) r;
+    switch (m_colorOrder) {
+      case kRGB -> {
+        m_buffer[index * 4] = (byte) b;
+        m_buffer[(index * 4) + 1] = (byte) g;
+        m_buffer[(index * 4) + 2] = (byte) r;
+      }
+      case kRBG -> {
+        m_buffer[index * 4] = (byte) g;
+        m_buffer[(index * 4) + 1] = (byte) b;
+        m_buffer[(index * 4) + 2] = (byte) r;
+      }
+      case kGBR -> {
+        m_buffer[index * 4] = (byte) r;
+        m_buffer[(index * 4) + 1] = (byte) b;
+        m_buffer[(index * 4) + 2] = (byte) g;
+      }
+      case kGRB -> {
+        m_buffer[index * 4] = (byte) b;
+        m_buffer[(index * 4) + 1] = (byte) r;
+        m_buffer[(index * 4) + 2] = (byte) g;
+      }
+      case kBRG -> {
+        m_buffer[index * 4] = (byte) g;
+        m_buffer[(index * 4) + 1] = (byte) r;
+        m_buffer[(index * 4) + 2] = (byte) b;
+      }
+      case kBGR -> {
+        m_buffer[index * 4] = (byte) r;
+        m_buffer[(index * 4) + 1] = (byte) g;
+        m_buffer[(index * 4) + 2] = (byte) b;
+      }
+      default -> {
+        // Unreachable
+      }
```

Comment:
According to https://docs.oracle.com/en/java/javase/17/language/switch-expressions.html this shouldn't be required for enums?

Reply:
Well, for some reason, a bunch of places in WPILib have default cases for enums anyway.

Reply:
Huh weird...

Reply:
This isn't a switch expression, it's still a statement.

Reply:
The empty default is functionally equivalent to not handling it